ingestion/src/metadata/ingestion/source/postgres.py

```diff
@@ -45,6 +47,19 @@ def get_connection_url(self):
 class PostgresSource(SQLSource):
     def __init__(self, config, metadata_config, ctx):
         super().__init__(config, metadata_config, ctx)
+        result = urlparse(self.connection_string)
+        username = result.username
+        password = result.password
+        database = result.path[1:]
+        hostname = result.hostname
+        port = result.port
+        self.pgconn = psycopg2.connect(
```

Comment:
No, that's the problem. Sqlalchemy maps these types to `NULL` and does not have an API to access them.

I thought about reusing the Sqlalchemy internal Postgres connection, but this is an implementation detail (hidden `_objects`) and they are stored in a pool so that's not very reliable.

Reply:
I will have a look at `engine.raw_connection()`

Reply:
I now use `engine.raw_connection()`, that's much cleaner!